html/browsers/the-window-object/named-access-on-the-window-object/navigated-named-objects.window.js

```diff
@@ -0,0 +1,49 @@
+[
+  {
+    "namedObject": "<div id=abc></div>",
+    "namedObjectLocalName": "div"
+  },
+  {
+    "namedObject": "<object name=abc></object>",
+    "namedObjectLocalName": "object"
+  },
+  {
+    "namedObject": "<iframe id=abc></iframe>",
+    "namedObjectLocalName": "iframe"
+  }
+].forEach(testData => {
+async_test(t => {
+  const frame = document.createElement("iframe");
+  t.add_cleanup(() => frame.remove());
+  frame.srcdoc = `<script>function f() { return abc }</script>${testData.namedObject}`;
+  let f, associatedAbc;
+  frame.onload = t.step_func(() => {
+    f = frame.contentWindow.f;
+    associatedAbc = f();
+    frame.onload = t.step_func_done(() => {
+      assert_equals(f(), associatedAbc);
+      assert_equals(associatedAbc.localName, testData.namedObjectLocalName);
+    });
+    frame.srcdoc = "<span id=abc></span>";
```

Comment:
Using srcdoc seems kind of like asking for extra complexity. Maybe duplicate this test with a normal `.src` setting?

Reply:
I'm not sure what you mean?